src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/smart-list.component.html

```diff
@@ -8,75 +8,85 @@
+                  <cd-alert-panel id="alert-self-test-passed"
+                                  size="slim"
+                                  type="info"
+                                  i18n-title
+                                  title="SMART overall-health self-assessment test result"
+                                  i18n>passed</cd-alert-panel>
+                </ng-container>
+                <ng-template #selfTestFailed>
+                  <cd-alert-panel id="alert-self-test-failed"
+                                  size="slim"
+                                  type="warning"
+                                  i18n-title
+                                  title="SMART overall-health self-assessment test result"
+                                  i18n>failed</cd-alert-panel>
+                </ng-template>
               </ng-template>
             </ng-template>
 
-            <ul ngbNav
-                #innerNav="ngbNav"
-                class="nav-tabs">
-              <li ngbNavItem>
-                <a ngbNavLink
-                   i18n>Device Information</a>
-                <ng-template ngbNavContent>
-                  <cd-table-key-value [renderObjects]="true"
-                                      [data]="device.value.info"></cd-table-key-value>
-                </ng-template>
-              </li>
-              <li ngbNavItem>
-                <a ngbNavLink
-                   i18n>SMART</a>
-                <ng-template ngbNavContent>
-                  <cd-table *ngIf="device.value.smart.attributes"
-                            [data]="device.value.smart.attributes.table"
-                            updateSelectionOnRefresh="never"
-                            [columns]="smartDataColumns"></cd-table>
-                  <cd-table-key-value *ngIf="device.value.smart.nvmeData"
-                                      [renderObjects]="true"
-                                      [data]="device.value.smart.nvmeData"
-                                      updateSelectionOnRefresh="never"></cd-table-key-value>
-                  <cd-alert-panel *ngIf="!device.value.smart.attributes && !device.value.smart.nvmeData"
-                                  type="info"
-                                  i18n>No SMART data available for this device.</cd-alert-panel>
-                </ng-template>
-              </li>
-            </ul>
+            <ng-container *ngIf="!isEmpty(device.value.info) || !isEmpty(device.value.smart)">
+              <ul ngbNav
+                  #innerNav="ngbNav"
+                  class="nav-tabs">
+                <li ngbNavItem>
+                  <a ngbNavLink
+                     i18n>Device Information</a>
+                  <ng-template ngbNavContent>
+                    <cd-table-key-value [renderObjects]="true"
+                                        [data]="device.value.info"></cd-table-key-value>
+                  </ng-template>
```

Comment:
Should we also show a info panel if this is not available?
on :57 we allow this to be empty.
And we are showing it for the SMART data.

Reply:
Thanks! Added the panel and corresponding tests.